src/controller/ExampleOperationalCredentialsIssuer.cpp

```diff
@@ -411,6 +411,88 @@ CHIP_ERROR ExampleOperationalCredentialsIssuer::GenerateNOCChain(const ByteSpan
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ExampleOperationalCredentialsIssuer::SignICACAfterValidation(const ByteSpan & icaCsr, MutableByteSpan & icac)
+{
+    uint8_t rcac[Credentials::kMaxDERCertLength];
+    MutableByteSpan rcacSpan = MutableByteSpan(rcac);
+    ChipDN rcac_dn;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    P256PublicKey pubkey;
+    ReturnErrorOnFailure(VerifyCertificateSigningRequest(icaCsr.data(), icaCsr.size(), pubkey));
+
+    uint16_t rcacBufLen = static_cast<uint16_t>(std::min(rcacSpan.size(), static_cast<size_t>(UINT16_MAX)));
+    PERSISTENT_KEY_OP(mIndex, kOperationalCredentialsRootCertificateStorage, key,
+                      err = mStorage->SyncGetKeyValue(key, rcacSpan.data(), rcacBufLen));
+    ReturnErrorOnFailure(err);
+
+    {
+        uint64_t rcacId;
+        // Found root certificate in the storage.
+        rcacSpan.reduce_size(rcacBufLen);
+        ReturnErrorOnFailure(ExtractSubjectDNFromX509Cert(rcacSpan, rcac_dn));
+        ReturnErrorOnFailure(rcac_dn.GetCertChipId(rcacId));
+        VerifyOrReturnError(rcacId == mRootIssuerId, CHIP_ERROR_INTERNAL);
+    }
+
+    ChipDN icac_dn;
+    ReturnErrorOnFailure(icac_dn.AddAttribute_MatterICACId(mIntermediateIssuerId));
+
+    ChipLogProgress(Controller, "Signing ICA CSR");
+
+    ReturnErrorOnFailure(
+        IssueX509Cert(mNow, mValidity, rcac_dn, icac_dn, CertType::kIcac, mUseMaximallySizedCerts, pubkey, mRootIssuer, icac));
+
+    VerifyOrReturnError(CanCastTo<uint16_t>(icac.size()), CHIP_ERROR_INTERNAL);
+
+    PERSISTENT_KEY_OP(mIndex, kOperationalCredentialsIntermediateCertificateStorage, key,
```

Comment:
No need to store as the cross-signed ICAC will be send back to jf-admin-app over RPC. jf-admin-app should be the one that saves the cross-signed ICAC (separate PR).

Reply:
This has been addressed in the latest commit.